security/sec-iride/src/main/java/org/geoserver/security/iride/IrideUserGroupService.java

```
@@ -0,0 +1,183 @@
+
+/**
+ * <code>GeoServer</code> user group security service, backed by <code>CSI</code> <code>IRIDE</code> service.
+ *
+ * @author "Simone Cornacchia - seancrow76@gmail.com, simone.cornacchia@consulenti.csi.it (CSI:71740)"
+ */
+public class IrideUserGroupService extends AbstractUserGroupService implements GeoServerUserGroupService {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOGGER = Logging.getLogger(IrideUserGroupService.class);
+
+    /**
+     * Constructor.
+     */
+    public IrideUserGroupService() {
+        super();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.geoserver.security.impl.AbstractGeoServerSecurityService#initializeFromConfig(org.geoserver.security.config.SecurityNamedServiceConfig)
+     */
+    @Override
+    public void initializeFromConfig(SecurityNamedServiceConfig config) throws IOException {
+        LOGGER.log(Level.INFO,
+            "Initializing {0} with configuration object: \n\t {1}",
+            new Object[] { this.getClass().getSimpleName(), config }
+        );
+
+        this.name = config.getName();
+
+        // TODO: check if needed and, if so, implement
+        super.initializeFromConfig(config);
+    }
+
+    /**
+     * <code>IRIDE</code> user group service is <em>read-only</em>, therefore there is no support for {@link GeoServerUserGroupStore}.
+     */
+    /*
+     * (non-Javadoc)
+     * @see org.geoserver.security.impl.AbstractGeoServerSecurityService#canCreateStore()
+     */
+    @Override
+    public boolean canCreateStore() {
+        return false;
+    }
+
+    /**
+     * <code>IRIDE</code> user group service is <em>read-only</em>, therefore there is no support for {@link GeoServerUserGroupStore}.
+     */
+    /*
+     * (non-Javadoc)
+     * @see org.geoserver.security.impl.AbstractUserGroupService#createStore()
+     */
+    @Override
+    public GeoServerUserGroupStore createStore() throws IOException {
+        return null;
```

If canCreateStore is false you don't need to override this method.

Returning null is not a good practice though. If it is not supported you should throw an UnsupportedException.